### PR TITLE
[CPU] SDPA supports Mixtral

### PR DESCRIPTION
**Pull request by luo-cheng2021**
### Details:
 - *SDPA supports Mixtral multi-query pattern*
 - *...*

### Tickets:
 - *[129812](https://jira.devtools.intel.com/browse/CVS-129812)*

### CODE REVIEW

File: src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp [hunk 1]
```diff
@@ -50,7 +58,31 @@ static std::shared_ptr<ov::Model> makeSDPA(const ov::PartialShape& inputShape, b
...
+                auto Gather_411049 = makeOP<opset8::Gather>({ShapeOf_250564_autogenerated, {2, 3}, 0}, {{"batch_dims", 0}});
+                auto ListConstruct_Concat_2 = makeOP<opset1::Concat>({ShapeOf_219297_autogenerated, {inputShape[1] / 4}, {4}, Gather_411049}, {{"axis", 0}});
+                auto expand_Abs = makeOP<opset1::Abs>({ListConstruct_Concat_2});
+                auto Constant_236480_autogenerated = makeConst(element::u8, ov::Shape({}), 0);
+                auto Broadcast_236481_autogenerated = makeOP<opset1::Broadcast>({{1.000000f},
+                    expand_Abs,
+                    Constant_236480_autogenerated}, {{"mode", "numpy"}});
+                auto expand_Broadcast = makeOP<opset1::Multiply>({unsqueeze_Unsqueeze_2,
+                    Broadcast_236481_autogenerated}, {{"auto_broadcast", "numpy"}});
+                auto ListConstruct_Concat_3 = makeOP<opset1::Concat>({ShapeOf_219297_autogenerated, {inputShape[1]}, Gather_411049}, {{"axis", 0}});
```
Comment (zhangYiIntel) [inline src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp:78]: Could we change the name with `autogenerated` postfix?

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp [hunk 2]
```diff
@@ -55,16 +55,25 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
...
+
+        auto constant_bcst = wrap_type<opset1::Constant>([] (Output<Node> output) -> bool {
+            auto node = std::dynamic_pointer_cast<opset1::Constant>(output.get_node_shared_ptr());
+            const auto& bcst_arg = node->cast_vector<float>();
+            return std::all_of(bcst_arg.begin(), bcst_arg.end(), [](float i) {
+                return i == 1.0;
+            });
+        });
+
+        auto computed_bcst = makePattern<opset1::Broadcast>({{1.000000f},
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp]: Done.

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp [hunk 3]
```diff
@@ -55,16 +55,25 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
 
     auto multi_query_bcst = [](std::shared_ptr<Node> kv) {
         auto reshape_kv = wrap_type<opset6::Reshape>({kv, any_input()});
-        auto unsqueeze_kv = makePattern<opset1::Unsqueeze>({kv, -2});
-        auto multiply_kv = wrap_type<opset6::Multiply>({reshape_kv | unsqueeze_kv, any_input()});
+        auto unsqueeze_kv = makePattern<opset1::Unsqueeze>({kv, any_input()});
+
+        auto constant_bcst = wrap_type<opset1::Constant>([] (Output<Node> output) -> bool {
+            auto node = std::dynamic_pointer_cast<opset1::Constant>(output.get_node_shared_ptr());
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp:61]: The lamda will be called after the `opset1::Constant` is matched, from the code: https://github.com/openvinotoolkit/openvino/blob/0e13b99a3d04da9fcf00cad349d8063cedd3bea1/src/core/src/pattern/op/wrap_type.cpp#L10-L18, it should be no need to check it again.

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp [hunk 4]
```diff
@@ -55,16 +55,25 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
...
+            const auto& bcst_arg = node->cast_vector<float>();
+            return std::all_of(bcst_arg.begin(), bcst_arg.end(), [](float i) {
+                return i == 1.0;
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp]: Done.

File: src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp [hunk 5]
```diff
@@ -50,7 +58,31 @@ static std::shared_ptr<ov::Model> makeSDPA(const ov::PartialShape& inputShape, b
         pastv = std::make_shared<ov::op::v8::Gather>(pastv, beam_idx, op::v0::Constant::create(element::i32, {1}, {0}));
         concatK = std::make_shared<ov::op::v0::Concat>(OutputVector{pastk, k}, 2);
         concatV = std::make_shared<ov::op::v0::Concat>(OutputVector{pastv, v}, 2);
-        sdp = std::make_shared<ov::opset13::ScaledDotProductAttention>(q, concatK, concatV, false);
+        if (hasMultiquery) {
+            auto make_multi_query = [&] (const Output<Node>& conat) {
+                auto ShapeOf_219297_autogenerated = makeOP<ov::op::TypeRelaxed<opset1::ShapeOf>>({beam_idx},
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp]: Done.

File: src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp (hunk 1 above)
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp]: Done.

File: src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp [hunk 6]
```diff
@@ -13,25 +13,33 @@
...
     auto q = std::make_shared<ov::op::v0::Parameter>(element::f32, inputShape);
-    auto k = std::make_shared<ov::op::v0::Parameter>(element::f32, inputShape);
-    auto v = std::make_shared<ov::op::v0::Parameter>(element::f32, inputShape);
+    auto kvInputShape = inputShape;
+    if (hasMultiquery) {
+        kvInputShape[1] = inputShape[1] / 4;
+    }
+    auto k = std::make_shared<ov::op::v0::Parameter>(element::f32, kvInputShape);
+    auto v = std::make_shared<ov::op::v0::Parameter>(element::f32, kvInputShape);
     auto init = std::make_shared<ov::op::v0::Parameter>(element::f32, inputShape);
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/tests/unit/transformations/state_concat_sdpa.cpp:36]: The new added code should not be related precision, it should be enough to use f32 to cover the function.

**Review by zhangYiIntel — APPROVED** [review]
LGTM thanks!

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**